Mathlib/LinearAlgebra/LinearProgramming.lean

```diff
@@ -0,0 +1,83 @@
+/-
+Copyright (c) 2023 Martin Dvorak. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Martin Dvorak
+-/
+import Mathlib.LinearAlgebra.Matrix.ToLin
+import Mathlib.LinearAlgebra.AffineSpace.AffineMap
+
+/-! # Linear programming
+
+Minimizing a linear function on a region defined by linear inequalities.
+
+## Main definitions
+
+ * `LinearProgram` defines a linear program in a form that generalizes "A x ≥ b".
+ * `LinearProgram.feasibles` is the set of all admissible solutions to given linear program.
+ * `LinearProgram.MinAt` defines an optimum solution of given linear program.
+
+-/
+
+/-- Linear program in the form of inequalities with general variables. -/
+structure LinearProgram (K : Type*) {V : Type*} (P : Type*)
+    [LinearOrderedField K] [AddCommGroup V] [Module K V] [AddTorsor V P] where
+  /-- Inequality constraints (given in the form "aᵀx - b ≥ 0") -/
+  constraints : List (P →ᵃ[K] K)
+  /-- The objective function (affine map) -/
+  objective : P →ᵃ[K] K
```

Comment:
I don't particularly like the design, with constraints and objective bundled together. The feasibility of constraints makes no mention of the objective, and the design should allow talking about this without an objective. Please separate these.

Reply:
Would you be more happy with the following definitions?
```
def LinearProgramFeas (R : Type*) {V : Type*} (P : Type*)
    [LinearOrderedRing R] [AddCommGroup V] [Module R V] [AddTorsor V P] :=
  List (P →ᵃ[R] R)

structure LinearProgramOpt (R : Type*) {V : Type*} (P : Type*)
    [LinearOrderedRing R] [AddCommGroup V] [Module R V] [AddTorsor V P] where
  constraints : LinearProgramFeas R P
  objective : P →ᵃ[R] R
```

Reply:
Why not two structures, one extending the other?

Presumably for the "just the constraints" version, you then want definitions `satisfiable_at` and `satisfiable`, with lots of lemmas about how these interact with list operations (e.g. adding/erasing etc constraints).